configure.ac

```diff
@@ -2800,7 +2841,16 @@ then
 		# to ensure that tests don't crash
 		# Note: This matches the value of THREAD_STACK_SIZE in
 		# thread_pthread.h
-		LINKFORSHARED="-Wl,-stack_size,1000000 $LINKFORSHARED"
+    stack_size="1000000"
+    if test "$with_ubsan" == "yes"
+    then
+        # Undefined behavior sanitizer requires an even deeper stack
+        stack_size="4000000"
+        AC_DEFINE_UNQUOTED(THREAD_STACK_SIZE,
+            0x$stack_size,
+            [Custom thread stack size to fit the undefined behavior sanitizer runtime.])
```

Comment:
I'd move the AC_DEFINE_UNQUOTED outside of the if body, that way setting THREAD_STACK_SIZE in Python/thread_pthread.h is no longer needed for macOS and the two values are automatically kept in sync.

Reply:
Note that confusingly LINKFORSHARED is used even if libpython is statically linked.